dusty/commands/validate.py

```diff
@@ -37,7 +37,19 @@ def _validate_lib_references(lib, specs):
     if not_present:
         raise ValidationException('Libs {} are not present in your specs'.format(', '.join(not_present)))
 
+def _check_name_overlap(specs):
+    apps = set(specs['apps'].keys())
+    libs = set(specs['libs'].keys())
+    services = set(specs['services'].keys())
+    app_service_overlap = apps.intersection(services)
+    if app_service_overlap:
+        raise ValidationException('Apps and services cannot share names: {}'.format(app_service_overlap))
+    app_lib_overlap = apps.intersection(libs)
+    if app_lib_overlap:
+        raise ValidationException('Apps and libs cannot share names: {}'.format(apps_lib_overlap))
```

Comment:
Well I'd say it makes sense to permit bundles to share names with apps.  We don't strictly need to check anything more than this to ensure functionality